### PR TITLE
To be published Oct 23, 2015

### PR DESCRIPTION
**Pull request by michaeldupuisjr**
Adds post about CPs with dynamic dependent keys.

### CODE REVIEW

File: source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+summary: "Need to determine the dependent key for your computed
+property at runtime?"
+published: false
+tags: ember, javascript
+---
+
+In Ember, computed properties allow you to declare functions as properties. More often than not, you're going to have a set of dependent keys which invoke a function when their values change:
+
+```js
+Team = Ember.Object.extend({
```
Comment (poteto) [inline source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md]: `const Team`?

File: source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  city: null,
+  league: null,
+  name: null,
+
+  fullName: Ember.computed('city', 'name', {
+    get() {
+      return `${this.get('city')} ${this.get('name')}`;
```
Comment (poteto) [inline source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md]: Maybe use the `Ember.get()` form instead, to keep things consistent?

File: source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+});
+
+export default Ember.Route.extend({
+  model() {
+    return {
+      teams: [
+        Team.create({ league: 'American', city: 'Boston', name: 'Red Sox' }),
+        Team.create({ league: 'American', city: 'New York', name: 'Yankees' }),
+        Team.create({ league: 'National', city: 'Los Angeles', name: 'Dodgers' }),
+        Team.create({ league: 'National', city: 'San Francisco', name: 'Giants' }),
```
Comment (poteto) [inline source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md]: Trailing comma! 🐎

File: source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  name: null,
+
+  fullName: Ember.computed('city', 'name', {
+    get() {
+      return `${this.get('city')} ${this.get('name')}`;
+    }
+  })
+});
+
+var redSox = Team.create({
```
Comment (poteto) [inline source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md]: `const`? :p

**Comment by poteto** [comment]
Nice! Maybe one thing you can add as well, is the scenario when you might have an arbitrary number of dependent keys. You can refer to the example in `getPropertiesByKeys` in our cpm util for our last client

File: source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+Keys"
+social: true
+author: "Michael Dupuis"
+twitter: michaeldupuisjr
+github: michaeldupuisjr
+summary: "Need to determine the dependent key for your computed
+property at runtime?"
+published: false
+tags: ember, javascript
+---
```
Comment (AaronSikes) [inline source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md]: could you add an ember_start_version line here?
either '1.13' or '2.0'

File: source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md (hunk 4 above)
Comment (michaeldupuisjr) [inline source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md]: Such is life without Ember Suave....

File: source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md (hunk 5 above)
Comment (michaeldupuisjr) [inline source/posts/2015-10-23-ember-best-practices-dynamic-dependent-keys-for-computed-properties.md]: No problem @AaronSikes

**Pull request merged**